test/module/irohad/ametsuchi/ametsuchi_mocks.hpp

```diff
@@ -336,7 +338,39 @@ namespace iroha {
                    bool(const shared_model::interface::BlocksQuery &));
     };
 
+    class MockTxPresenceCahce : public iroha::ametsuchi::TxPresenceCache {
+      MOCK_CONST_METHOD1(check,
+                         iroha::ametsuchi::TxCacheStatusType(
+                             const shared_model::crypto::Hash &hash));
+
+      MOCK_CONST_METHOD1(
+          check,
+          iroha::ametsuchi::TxPresenceCache::BatchStatusCollectionType(
+              const shared_model::interface::TransactionBatch &));
+    };
+
   }  // namespace ametsuchi
 }  // namespace iroha
 
+namespace std {
```

Comment:
overloading things in std namespace is undefined behavior. If you want google test to print your values, please refer to: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#teaching-googletest-how-to-print-your-values

It has 2 solutions: define << operator, or PrintTo function. I would prefer PrintTo as it is more google test specific.

However if you decide to overload <<, then do it in the same namespace as the classes you have declared.